src/org/impact2585/frc2016/systems/AccelerometerSystem.java

```diff
@@ -36,17 +114,112 @@ public void init(Environment environ) {
 	 */
 	@Override
 	public void run() {
-		I2Cbus.read(DEVICE_ADDRESS, 14, buffer);
-		int j = 0;
-		//raw values are ax, ay, az, gx, gy, gz
-		for(int i = 0; i< rawValues.length; i++) {
-			if(j == 3)
-				j++;
-			rawValues[i] = (short) ((((short)buffer[j * 2]) << 8) | buffer[j * 2 + 1]);
-			j++;
+		//wait for the FIFO count to reach the dmp packet size
+		while(currentFIFOcount < DMP_PACKET_SIZE) currentFIFOcount = getFIFOcount();
```

Comment:
It may be a bad idea to use a while loop like that in our system. The code needs to iterate every 20 ms and while loops like this may make things stall. Instead, it might be a better idea to make the `run` method do nothing other than update the `currentFIFOcount` instance variable if the FIFO count is less than the dmp packet size.